src/dataframe/dataframe.jl

```diff
@@ -596,19 +600,118 @@ Base.setindex!(df::DataFrame, x::Void, col_ind::Int) = delete!(df, col_ind)
+"""
+Insert a column into a data frame in place.
+
+
+```julia
+insert!(df::DataFrame, col_ind::Int, item::AbstractVector, name::Symbol;
+        makeunique::Bool=false)
+```
+
+### Arguments
+
+* `df` : the DataFrame to which we want to add a column
+
+* `col_ind` : a position at which we want to insert a column
+
+* `item` : a column to be inserted into `df`
+
+* `name` : column name
+
+* `makeunique` : Defines what to do if `name` already exists in `df`;
+  if it is `false` an error will be thrown; if it is `true` a new unique name will
+  be generated by adding a suffix
+
+### Result
+
+* `::DataFrame` : a `DataFrame` with added column.
+
+### Examples
+
+```jldoctest
+julia> d = DataFrame(a=1:3)
+3×1 DataFrames.DataFrame
+│ Row │ a │
+├─────┼───┤
+│ 1   │ 1 │
+│ 2   │ 2 │
+│ 3   │ 3 │
+
+julia> insert!(d, 1, 'a':'c', :b)
+3×2 DataFrames.DataFrame
+│ Row │ b   │ a │
+├─────┼─────┼───┤
+│ 1   │ 'a' │ 1 │
+│ 2   │ 'b' │ 2 │
+│ 3   │ 'c' │ 3 │
+```
+
+"""
+function Base.insert!(df::DataFrame, col_ind::Int, item::AbstractVector, name::Symbol;
+                      makeunique::Bool=false)
     0 < col_ind <= ncol(df) + 1 || throw(BoundsError())
     size(df, 1) == length(item) || size(df, 1) == 0 || error("number of rows does not match")
 
+    if haskey(df, name)
+        if makeunique
+            k = 1
+            while true
+                # we only make sure that new column name is unique
+                # if df originally had duplicates in names we do not fix it
+                nn = Symbol("$(name)_$k")
+                if !haskey(df, nn)
+                    name = nn
+                    break
+                end
+                k += 1
+            end
+        else
+            # TODO: remove depwarn and uncomment ArgumentError below
+            Base.depwarn("Inserting duplicate column name is deprecated, use makeunique=true.", :insert!)
+            # msg = """Duplicate variable name $(name).
+            #      Pass makeunique=true to make it unique using a suffix automatically."""
+            # throw(ArgumentError(msg))
+        end
+    end
     insert!(index(df), col_ind, name)
     insert!(df.columns, col_ind, item)
     df
 end
 
-function Base.insert!(df::DataFrame, col_ind::Int, item, name::Symbol)
-    insert!(df, col_ind, upgrade_scalar(df, item), name)
+function Base.insert!(df::DataFrame, col_ind::Int, item, name::Symbol; makeunique::Bool=false)
+    insert!(df, col_ind, upgrade_scalar(df, item), name, makeunique=makeunique)
 end
 
+"""
+Merge data frames.
+
+
+```julia
+merge!(df::DataFrame, others::AbstractDataFrame...)
+```
+
+For every column `c` with name `n` in `others` sequentially perform `df[n] = c`.
+This behavior is identical with how `merge!` works for any `Associative` type.
+Use `join` if you want to join two `DataFrame`s.
+
+**Arguments**
+
+* `df` : the DataFrame to merge into
+* `others` : `AbstractDataFrame`s to be merged into `df`
```

Comment:
`makeunique` should be mentioned.

Reply:
OK

Reply:
Sorry - `merge` does not have `makeunique` argument. It follows `Associative` interface, i.e. it overwrites duplicate columns. I have added an additional explanation.

Reply:
But I have added information about `makeunique` to `DataFrame` docstring.

Reply:
Funny, not sure how I got confused.